homeassistant/components/withings/common.py

```diff
@@ -335,3 +338,15 @@ def get_data_manager(
     )
 
     return dm_dict[entry_id]
+
+
+class WithingsLocalOAuth2Implementation(LocalOAuth2Implementation):
+    """Oauth2 implementation that only uses the external url."""
+
+    @property
+    def redirect_uri(self) -> str:
+        """Return the redirect uri."""
+        url = get_url(
+            self.hass, allow_internal=False, prefer_cloud=True, prefer_external=True
```

Comment:
Prefer external is now redundant, as internal is disabled already.

```suggestion
            self.hass, allow_internal=False, prefer_cloud=True
```